pkg/kubelet/cm/devicemanager/plugin/v1beta1/handler.go

```diff
@@ -87,12 +91,23 @@ func (s *server) disconnectClient(name string, c Client) error {
 	return c.Disconnect()
 }
 
-func (s *server) registerClient(name string, c Client) {
+func (s *server) registerClient(name string, c Client) error {
 	s.mutex.Lock()
 	defer s.mutex.Unlock()
 
+	if existing, ok := s.clients[name]; ok {
+		path1 := existing.(DevicePlugin).SocketPath()
+		path2 := c.(DevicePlugin).SocketPath()
```

Comment:
do we have a way to get the path avoiding the cast?

Reply:
There‘s no other way except using `client{}` structure. The `Interface` way is more recommended in the `Go` world.

Reply:
I'd try adding a `SocketPath` method to the `Client` interface and evaluate the impacts.

Reply:
or, in general, carrying this information around somehow. I'd really like to explore options to avoid the cast.

Reply:
> I'd try adding a SocketPath method to the Client interface and evaluate the impacts.

I added it in this PR.